### ISSUE TITLE
Fix false negatives for TypeScript files and Babel config

### ISSUE THREAD

**Issue by Gruzchick** [stylelint/stylelint#4732]
> Clearly describe the bug

Stylelint do not work with .tsx files with styled components from emotion library

> What code is needed to reproduce the bug?

https://github.com/Gruzchick/bugreport

execute "yarn lint:css" in the terminal

then you will see errors from src/components/App/styles.css
but errors from src/components/App/styles.tsx will do not shown

> What did you expect to happen?

Errors from src/components/App/styles.tsx

**Comment by jeddy3** [comment]
@Gruzchick Thanks for the report and for using the template.

I was able to reproduce the issue using your repo. If I remove the `.babelrc` file it works correctly. There was a fix for this (https://github.com/stylelint/postcss-css-in-js/pull/2), but it appears to have resurfaced.

It's likely a problem with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a growing list of the problems with the syntax that we need help to fix.

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time.

**Comment by sobolev-ax** [comment]
@jeddy3 Hello
I also tried use 11 version in which was resolved the same bug. But it still doesn't work. Are there some workaround?
https://github.com/stylelint/stylelint/blob/HEAD/CHANGELOG.md#1100

**Comment by jeddy3** [comment]
> Are there some workaround?

Not that I'm aware of. The problem needs to be fixed in the parser. Please consider contributing to [the parser](https://github.com/stylelint/postcss-css-in-js) if you have time.

**Comment by kondei** [comment]
:memo: I have the same problem from stylelint@13.3.1

When "plugins" array in .babelrc file has elements more than 1.

**Comment by vault-developer** [comment]
>  I have the same problem from stylelint@13.3.1
> 
> When "plugins" array in .babelrc file has elements more than 1.

I had similar problem with stylelint@13.9.0. 
The thing was that I forgot to add in my package.json plugins, which was included in .babelrc.

**Comment by kondei** [comment]
> 📝 I have the same problem from stylelint@13.3.1
> 
> When "plugins" array in .babelrc file has elements more than 1.

I may have found a workaround for my problem.

Adding

```
  "parserOpts": {
    "plugins": ["jsx", "typescript"]
  },
```

in babel setting made it work at stylelint 13.12.0

**Comment by christophehurpeau** [comment]
I can confirm the workaround from @kondei works for me. To apply only when running stylelint, this is what my `babel.config.js` file looks like:

```
module.exports = {
  presets: [ ... ]
};

if (process.env._ && process.env._.endsWith('/stylelint')) {
  module.exports.parserOpts = {
    plugins: ['jsx', 'typescript'],
  };
}
```

**Comment by marwan38** [comment]
Facing this issue as well, I'm surprised it's been around this long with the fix not being patched.

**Comment by jeddy3** [comment]
> I'm surprised it's been around this long with the fix

The problem is with the upstream parser. Unfortunately, it has issues:
- it monkey-patches PostCSS@7
- it's monolithic and tries to support every CSS-in-JS flavour
- the original author is no longer active on GitHub

> @jeddy3 Would this be the correct place to look at?

As such, the parser [will be removed](https://github.com/stylelint/stylelint/issues/5289#issuecomment-840772900) in the [next major release of stylelint](https://github.com/stylelint/stylelint/issues/5205). It'll be replaced by leaner parsers that are specific to the shape of the CSS-in-JS library it supports. 

If you'd like to help out, please consider investigating what a parser for your CSS-in-JS library of choice may look like. See https://github.com/stylelint/stylelint/issues/5205#issuecomment-856662626.

> I just moved my babel config from the root of the application to somewhere elses (process.cmd()/config/babel.config)

Thanks for sharing your workaround!

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
